src/transaction.js

```diff
@@ -138,13 +144,111 @@ Transaction.fromBuffer = function (buffer, network = networks.bitcoin, __noStric
     }
   }
 
+  function readZKProof () {
+    var zkproof
+    if (tx.version >= Transaction.ZCASH_SAPLING_VERSION) {
+      zkproof = {
+        sA: readSlice(48),
+        sB: readSlice(96),
+        sC: readSlice(48)
+      }
+    } else {
+      zkproof = {
+        gA: readCompressedG1(),
+        gAPrime: readCompressedG1(),
+        gB: readCompressedG2(),
+        gBPrime: readCompressedG1(),
+        gC: readCompressedG1(),
+        gCPrime: readCompressedG1(),
+        gK: readCompressedG1(),
+        gH: readCompressedG1()
+      }
+    }
+    return zkproof
+  }
+
+  function readJoinSplit () {
```

Comment:
It looks like JoinSplits can be encoded in two different ways depending on version (v2 + v3 vs v4). Is that handled?